src/server/server_family.cc

```diff
@@ -1184,27 +1181,40 @@ GenericError ServerFamily::DoSave(bool new_version, string_view basename, Transa
   if (new_version) {
     // In the new version (.dfs) we store a file for every shard and one more summary file.
     // Summary file is always last in snapshots array.
-    snapshots.resize(shard_set->size() + 1);
+    const size_t sz = shard_set->size();
+    constexpr const string_view ext = ".dfs.tmp"sv;
+    snapshots.resize(sz + 1);
+
+    // Set file names for shards
+    for (auto sid = 0u; sid < sz; ++sid) {
+      auto& filename = snapshots[sid].second;
```

Comment:
This assignment seems redundant. It appears that you overwrite `filename` on the next statement below.

Reply:
just saves some writing ;)